lib/chef/knife/ec2_server_create.rb

```diff
@@ -753,17 +719,13 @@ def plugin_validate_options!
           end
         end
 
-        if config_value(:tag_node_in_chef)
-          ui.warn("[DEPRECATED] --tag-node-in-chef option is deprecated. Use --chef-tag option instead.")
-        end
-
-        if config_value(:tags)
-          ui.warn("[DEPRECATED] --tags option is deprecated. Use --aws-tag option instead.")
+        if config_value(:chef_tag)
+          ui.warn("[DEPRECATED] --chef-tag option is deprecated and will be removed in future release. Use --tags TAGS option instead.")
```

Comment:
I think `deprecated_options` could be used to handle this

Reply:
@Nimesh-Msys As `--chef-tag` is used multiple times for multiple values and proc helper is used to storing these values in the array. 
But  `deprecated_option` don't have any proc option as of now.

We might think of improving mixlib-cli to support `value_mapper` with  `proc` block as well.